src/intTest/java/uk/nhs/digital/nhsconnect/lab/results/mesh/MeshClientIntegrationTest.java

```diff
@@ -72,7 +71,7 @@ void when_callingMeshSendMessageEndpoint_expect_messageIdIsReturned() {
     void when_callingMeshGetMessageEndpoint_expect_messageIsReturned() {
         final MeshMessageId testMessageId = getMeshClient().sendEdifactMessage(OUTBOUND_MESH_MESSAGE);
 
-        final InboundMeshMessage meshMessage = getLabResultsMeshClient().getEdifactMessage(testMessageId.getMessageID());
+        final var meshMessage = getLabResultsMeshClient().getEdifactMessage(testMessageId.getMessageID());
```

Comment:
Does it not become a bit confusing in this case what's the type of the message? (P.S. maybe it's just for me) But also, just a general question, why would we tend to use the var instead?

Reply:
I don't, but if you do then others will do and we can look for an alternative.

However, I would encourage you to look at this test in your IDE and see if it's unclear. Remember it's an integration test, so all we care about is that the overall flow of the system is functional. By using `var` here the statement changes from:

"We expect the mesh client to be able to retrieve an `InboundMeshMessage` with content X and workflowId Y"

to:

"We expect the mesh client to be able to retrieve a mesh message [because of the variable name] with content X and workflowId Y"

As for the general question, I've said before I find 2 scenarios make the use of `var` viable (but never mandatory):
1. When the type of a variable is obvious from its name, its derivation, and/or its context
2. When the type of a variable is unimportant, for example when it's simply being passed through a method without being inspected or manipulated.

If a variable has a particularly complicated type (`Map<String, List<Map<Integer, Set<Person>>>>` or something) then readability might be improved by just hiding it all behind `var`. A bit of code should be understandable very quickly - so spending time trying to understand the complex type could be seen as a distraction from understanding what the code is actually trying to accomplish.

Reply:
Thanks a lot for your very detailed explanation, @andrewwa-kainos! In this case, yes, it would be good to keep var. However, I also agree with the Bartosz that we can change variable name to inboundMeshMessage